Sources/PostgreSQL/Connection.swift

```diff
@@ -118,18 +119,63 @@ public final class Connection: ConnectionProtocol {
     }
 
     public func open() throws {
-        connection = PQsetdbLogin(
-            connectionInfo.host,
-            String(connectionInfo.port),
-            connectionInfo.options ?? "",
-            connectionInfo.tty ?? "",
-            connectionInfo.databaseName,
-            connectionInfo.username ?? "",
-            connectionInfo.password ?? ""
-        )
-
-        if let error = mostRecentError {
-            throw error
+        guard connection == nil else {
+            throw ConnectionError(description: "Connection already opened.")
+        }
+        
+        var components = URLComponents()
```

Comment:
Personally just think that is a nit, it's a local scope var and mutated either way step by step…